### DISCUSSION TITLE
Overall Problem with Video Download Helper?

### DISCUSSION THREAD

**Discussion by jonni22** [aclap-dev/video-downloadhelper#1094] [Help and Q&A]
### Your question

so I downloaded the Video Downloader Browser extension for edge a couple of weeks ago and used it to download tons of movies and even thought about upgrading to premium, but now since a few days the downloads get really really slow and eventually freeze or even stop themselves. I tried different sites and different videos but it's always the same and even if downloading just one video at a time (before the problems started, I downloaded 5-8 items at a time effortlessly) 
any suggestions what's the problem? is there an overall problem at the moment? do I have to update something?

### Addon & CoApp version and details

version: 9.0.2.12
target: microsoft
channel: stable
lang: de
coapp: {"found":true,"new_version":false,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19"}
license: {"unset":true}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36 Edg/127.0.0.0
{
    "used_sidebar_button": true,
    "used_popup_button": true,
    "concurrent_downloads_max": 15,
    "view_options": {
        "all_tabs": false,
        "hide_downloaded": false,
        "low_quality": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "sort_by_status": true,
        "sort_reverse": false
    },
    "use_sidebar": true,
    "open_count_store": 1,
    "successfull_dl": 27,
    "never_show_successfull_dl_message": true
}

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

It's good that you have not bought a license yet.  You are doing the right thing by testing the software to see whether it works before you buy a license.  Buying a license does not enable new features in VDH nor does it make VDH work on certain web sites that don't work when you don't have a license.

Download speed is such a tricky subject.  There are so many variables involved.  But I could compare results with you.  I'm running licensed VDH for Firefox.  If I had a few (don't go overboard) of the URLs that are going slow for you, I could try downloading them & see what speed I get.  This would be URLs of specific videos, not just the name of a web site.  The one caveat is that I will not register for a site that requires you to do that before it will let you download their content.  This has to be a free site (or sites) that is (are) generally available.

If you don't have a license, VDH will write a watermark into the video . . . under certain circumstances.  That can be a lengthy step, sometimes taking hours for just a short video.  This should not be confused with download speed.  The video is downloaded, then the watermark is written as a subsequent step.  We need to be clear that we are talking about the true speed of data download over your Internet connection & not lumping in the writing of the watermark.

**Comment by paulrouget** [comment]
I need one URL video that is slow, otherwise I can't help.

**Reply by jonni22** [reply]
It's with every movie/video from https://ww4.fmovies.co/24/ and https://gostream.to  for example

**Reply by GrampaWildWilly** [reply]
I very explicitly asked you to give me a URL of a movie, not the URL of a web site.  And what do you do?  The idea was for you to download a movie & then for me to download the same movie.  Apparently, that's just too much for you to grasp.

OK.  I'll set the target & you download that same movie so we can compare results.  I went to Oppenheimer (because that was the movie at the top of the home page when I visited fmovies) & told VDH to download the movie.  You're right.  The download speed was something like 10,000 bytes per second.  Utterly putrid.  After about 5 minutes crawling along like that, Firefox crashed, asking me to send a report to Mozilla.  Which I did.  I didn't bother to even look at it before I sent it.  I hope they find it useful.

So I went to Plan B.  I used the Developer Tools of Firefox to find the HLS master manifest for this & constructed an invocation of ffmpeg to download this.  I note that there is Timed_ID3 data in this movie, something that used to cause a lot of trouble for VDH but no longer does.  At least, it's not supposed to.  VDH uses ffmpeg to download content now, & ffmpeg is incapable of handling Timed_ID3 data.  By incapable I mean ffmpeg just throws up an error message saying it can't recognize the Timed_ID3 data & ignores it.  So that is what is supposed to happen inside any download via VDH.  That is what happened when I did the download directly in ffmpeg.

But ffmpeg directly executed by me without reference to VDH also got the same 10,000 bytes per second download speed.  After a bit, I just cancelled the ffmpeg session.  Here's the log I got.  Note that ffmpeg invokes ffprobe to start, so I did not include a separate listing of my ffprobe that I did before I launched ffmpeg.

[Oppenheimer mp4 Log.txt](https://github.com/user-attachments/files/16495331/Oppenheimer.mp4.Log.txt)

I was astonished to see that I apparently had a bit over 2 hours of this 3 hour movie.  You can see in the ffmpeg log that it started off at a speed factor that was well less than 1.  It got as low as 0.015, which means that download would have taken about 3/0.015 = 200 hours to download.  This was reflected in the Resource Monitor as a download speed of under 10,000 bytes per second.  But then it got a corrupt packet & that seemed to wake things up.  The speed factors gradually increased to the level of 10.  That would calculate out to a download lasting about 18 minutes.  I was probably too hasty in cancelling this download.  The Resource Monitor & ffmpeg did not seem to be in synch on the subject of download speed.  I checked the file as I had downloaded it by sampling it quickly at 1-minute intervals.  Indeed, I had what it says in the ffprobe report at the end of that log: a complete movie of duration 2:16:21.  By complete I mean I had what ffprobe claimed I had.  It wasn't the complete movie because I interrupted the download before it was finished.  But everything that ffmpeg said it had downloaded indeed was here.

Well, I had to retry it.  This time, Resource Monitor was showing me proper download speeds.  They gradually increased to an equilibrium state somewhere in the 23-24 million bytes per second range.  You can see that this time, ffmpeg was getting speed factors that climbed up over 90.  This log shows just how fast things went.  Look for the little report of download duration, download file size, & average download speed between the ffmpeg log & the ffprobe at the bottom.  You'll note the video bit rates for this movie are actually quite low, which is why a 3-hour movie could download this fast.  Mind you, the audio bit rate is exceptionally high.

[Oppenheimer mp4 Log.txt](https://github.com/user-attachments/files/16495589/Oppenheimer.mp4.Log.txt)

This time, a quick sampling of the movie showed I had every second of it, right to the last end credit.  There was also audio all the way to the end.

Well, if I was willing to retry it in ffmpeg, I had to retry it in VDH.  Sure enough, VDH on my second attempt was getting a bit over 20 million bytes per second service.  That's only slightly less than what ffmpeg got.

But I don't think you can compare VDH against "manual" ffmpeg.  I think you need to look at how this web site behaved.  One minute I can hardly get it to work.  It was giving me a download speed that would have been putrid in the days of dial-up modems over a telephone line.  A few minutes later, it was giving me exceptionally wonderful download speed.  The conclusion is that the web site is temperamental.  If you see a bad speed, just cancel the download & try again.  Rinse & repeat until you see you are getting good service.  My experience clearly points to the web site being the culprit & not VDH (or ffmpeg) at all.  You don't like this advice?  Well, that's your choice.  But this is the advice I cannot escape offering based on my experience.

**Reply by GrampaWildWilly** [reply]
Oops.  Forgot to mention my operating environment.  It's Windows 11 Pro 64-bit version 23H2 build 22631.3958, Firefox 128.0.3 64-bit, licensed VDH 9.1.0.41 beta, CoApp 2.0.19.  The ffmpeg I used in my non-VDH downloads is the one supplied with CoApp 2.0.19.  Which is to say that I should get the same results with VDH & with ffmpeg directly.  There are examples on here in which that is not the case but it was the case here.
